packages/field/src/RegisterFormElementMixin.js

```diff
@@ -0,0 +1,59 @@
+import { dedupeMixin } from '@lion/core';
+import { ObserverMixin } from '@lion/core/src/ObserverMixin.js';
+
+/**
+ * #RegisterFormElementMixin :
+ *
+ * This Mixin registers form elements, it's applied on:
+ * - FormControlMixin (which is extended to LionField, LionInput, LionTextarea, LionSelect etc. etc.)
+ *
+ * @polymerMixin
+ * @mixinFunction
+ */
+export const RegisterFormElementMixin = dedupeMixin(
+  superclass =>
+    // eslint-disable-next-line no-shadow, no-unused-vars
+    class RegisterFormElementMixin extends ObserverMixin(superclass) {
+      connectedCallback() {
+        super.connectedCallback();
+        this._registerFormElement();
+        this._requestParentFormGroupUpdateOfResetModelValue();
+      }
+
+      /**
+       * Fires a registration event in the next frame.
+       *
+       * Why next frame?
+       * if ShadyDOM is used and you add a listener and fire the event in the same frame
+       * it will not bubble and there can not be caught by a parent element
+       * for more details see: https://github.com/Polymer/lit-element/issues/658
+       * will requires a `await nextFrame()` in tests
+       */
+      _registerFormElement() {
+        this.updateComplete.then(() => {
+          this.dispatchEvent(
+            new CustomEvent('form-element-register', {
+              detail: { element: this },
+              bubbles: true,
+            }),
+          );
+        });
+      }
+
+      /**
+       * Makes sure our parentFormGroup has the most up to date resetModelValue
+       * FormGroups will call the same on their parentFormGroup so the full tree gets the correct
+       * values.
+       *
+       * Why next frame?
+       * @see {@link this._registerFormElement}
+       */
+      _requestParentFormGroupUpdateOfResetModelValue() {
+        this.updateComplete.then(() => {
+          if (this.__parentFormGroup) {
```

Comment:
```suggestion
          if (this.__parentFormGroup && this.__parentFormGroup._updateResetModelValue) {
```

Reply:
we probably also want to add a test for it?

Reply:
Ah, yes. let me look into it.